File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        
        int num = 1;
        int j=1;

        for(int i=1;i<nums.size();i++,j++){

            if(num==2){
                while(i < nums.size() && nums[i]==nums[i-1]){
                    i++;
                }
                num = 1;
                if(i==nums.size()){
                    break;
                }
            }

            if(nums[i]==nums[i-1]){
                num++;
            }
            nums[j] = nums[i];
        }
        return j;
    }
};

File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int curnum = nums[0];
        int count = 1;
        int left=1,right=1;
        for(;right<nums.length;right++){
            if(nums[right]==curnum){
                count++;
            }else{
                curnum = nums[right];
                count = 1;
            }
            nums[left] = nums[right];

            if(count<=2){
                left++;
            }
        }
        return left;
    }
}